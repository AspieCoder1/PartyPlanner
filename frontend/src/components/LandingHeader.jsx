"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var LandingHeader_module_scss_1 = require("./LandingHeader.module.scss");
var LandingHeader = /** @class */ (function (_super) {
    __extends(LandingHeader, _super);
    function LandingHeader() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    LandingHeader.prototype.render = function () {
        return (<header className={LandingHeader_module_scss_1.default.header}>
        <div className={LandingHeader_module_scss_1.default.container}>
          <h1 className={LandingHeader_module_scss_1.default.title}>PartyPlanner.io</h1>
          <div className={LandingHeader_module_scss_1.default.headerButtons}>
            <button onClick={function () { console.log("Login button"); }} className={LandingHeader_module_scss_1.default.loginButton}>Login</button>
            <button onClick={function () { console.log("Register button"); }} className={LandingHeader_module_scss_1.default.registerButton}>Register</button>
          </div>
        </div>
      </header>);
    };
    return LandingHeader;
}(React.Component));
exports.default = LandingHeader;
