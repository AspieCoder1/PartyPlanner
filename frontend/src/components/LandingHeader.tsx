import * as React from 'react';
import styles from './LandingHeader.module.scss';

type IProps = {
  onClickLogin: () => void;
  onClickRegister: () => void;
}

export default class LandingHeader extends React.Component<IProps> {
  render() {
    return (
      <header className={styles.header}>
        <div className={styles.container}>
          <h1 className={styles.title}>PartyPlanner.io</h1>
          <div className={styles.headerButtons}>
            <button onClick={this.props.onClickLogin} className={styles.loginButton}>Login</button>
            <button onClick={this.props.onClickRegister} className={styles.registerButton}>Register</button>
          </div>
        </div>
      </header>
    );
  }
}
