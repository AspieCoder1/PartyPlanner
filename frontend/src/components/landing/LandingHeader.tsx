import * as React from 'react';
import styles from './LandingHeader.module.scss';
import logo from '../../img/partyplanner.png';

type IProps = {
	onClickLogin: () => void;
	onClickRegister: () => void;
};

export default class LandingHeader extends React.Component<IProps, never> {
	render(): React.ReactNode {
		return (
			<header className={styles.header}>
				<div className={styles.container}>
					<div className={styles.title}>
						<img src={logo} alt='PartyPlanner.io' />
					</div>
					<div className={styles.headerButtons}>
						<button onClick={this.props.onClickLogin} className={styles.loginButton}>
							Login
						</button>
						<button onClick={this.props.onClickRegister} className={styles.registerButton}>
							Register
						</button>
					</div>
				</div>
			</header>
		);
	}
}
