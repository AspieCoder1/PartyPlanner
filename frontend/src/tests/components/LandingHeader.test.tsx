import * as React from 'react';
import * as renderer from 'react-test-renderer';
import LandingHeader from '../../components/LandingHeader';

describe('Landing page component', () => {
	it('Should render component correctly', () => {
		const tree = renderer.create(<LandingHeader  onClickLogin={() => {}} onClickRegister={() => {}} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
