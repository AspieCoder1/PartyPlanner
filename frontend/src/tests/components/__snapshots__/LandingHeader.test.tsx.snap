// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component Should render component correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="title"
    >
      <img
        alt="PartyPlanner.io"
        src="partyplanner.png"
      />
    </div>
    <div
      className="headerButtons"
    >
      <button
        className="loginButton"
        onClick={[Function]}
      >
        Login
      </button>
      <button
        className="registerButton"
        onClick={[Function]}
      >
        Register
      </button>
    </div>
  </div>
</header>
`;
